src/parse/mod.rs

```diff
@@ -0,0 +1,1937 @@
+                    self.whitespace();
+                    params.push(' ');
+                    continue;
+                }
+                _ => {}
+            }
+            params.push(tok.kind);
+        }
+
+        let raw_body = self.parse()?;
+        let mut rules = Vec::with_capacity(raw_body.len());
+        let mut body = Vec::new();
+
+        for stmt in raw_body {
+            match stmt {
+                Stmt::Style(..) => body.push(stmt),
+                _ => rules.push(stmt),
+            }
+        }
+
+        if !self.super_selectors.last().is_empty() {
+            body = vec![Stmt::RuleSet {
+                selector: self.super_selectors.last().clone(),
+                body,
+                super_selector: Selector::new(),
+            }];
+        }
+
+        body.append(&mut rules);
+
+        Ok(Stmt::UnknownAtRule {
+            name,
+            super_selector: Selector::new(),
+            params: params.trim().to_owned(),
+            body,
+        })
+    }
+
+    fn import(&mut self) -> SassResult<Vec<Stmt>> {
+        self.whitespace();
+        let mut file_name = String::new();
+        let next = match self.toks.next() {
+            Some(v) => v,
+            None => todo!("expected input after @import"),
+        };
+        match next.kind {
+            q @ '"' | q @ '\'' => {
+                file_name.push_str(
+                    &self
+                        .parse_quoted_string(q)?
+                        .node
+                        .unquote()
+                        .to_css_string(self.span_before)?,
+                );
+            }
+            _ => return Err(("Expected string.", next.pos()).into()),
+        }
+        if let Some(t) = self.toks.peek() {
+            if t.kind == ';' {
+                self.toks.next();
+            }
+        }
+
+        self.whitespace();
+
+        let path: &Path = file_name.as_ref();
+
+        let mut rules = Vec::new();
+        let path_buf = if path.is_absolute() {
+            // todo: test for absolute path imports
+            path.into()
+        } else {
+            self.path
+                .parent()
+                .unwrap_or_else(|| Path::new(""))
+                .join(path)
+        };
+        // todo: will panic if path ended in `..`
+        let name = path_buf.file_name().unwrap();
+        if path_buf.extension() == Some(OsStr::new(".css")) {
+            // || name.starts_with("http://") || name.starts_with("https://") {
+            todo!("css imports")
+        }
+        let mut p1 = path_buf.clone();
+        p1.push(OsString::from("index.scss"));
+        let mut p2 = path_buf.clone();
+        p2.push(OsString::from("_index.scss"));
+        let paths = [
+            path_buf.with_file_name(name).with_extension("scss"),
+            path_buf.with_file_name(format!("_{}.scss", name.to_str().unwrap())),
+            path_buf,
+            p1,
+            p2,
+        ];
```

Comment:
Why not? (not tested)
```suggestion
        let paths = [
            path_buf.with_file_name(name).with_extension("scss"),
            path_buf.with_file_name(format!("_{}.scss", name.to_str().unwrap())),
            path_buf,
            path_buf.join("index.scss"),
            path_buf.join("_index.scss"),
        ];
```

Reply:
Good point!